packages/firestore/src/api/database.ts

```diff
@@ -287,14 +295,20 @@ export function configureFirestore(firestore: Firestore): void {
     settings.cache?._offlineComponentProvider &&
     settings.cache?._onlineComponentProvider
   ) {
-    firestore._firestoreClient.uninitializedComponentsProvider = {
-      offline: settings.cache._offlineComponentProvider,
-      online: settings.cache._onlineComponentProvider
+    firestore._firestoreClient._uninitializedComponentsProvider = {
+      _offlineKind: settings.cache.kind,
+      _offline: settings.cache._offlineComponentProvider,
+      _online: settings.cache._onlineComponentProvider
     };
   }
 }
 
 /**
+ * NOTE: this function will be deprecated in a future major release. Instead, set
```

Comment:
Consider using the `@deprecated` tag. https://api-extractor.com/pages/tsdoc/tag_deprecated/